src/liveEditor/generators/generateToolbar.tsx

```diff
@@ -66,8 +67,21 @@ export function appendFieldPathDropdown(
         targetElementDimension.left - LIVE_PREVIEW_OUTLINE_WIDTH_IN_PX;
     const adjustedDistanceFromLeft = Math.max(distanceFromLeft, TOOLBAR_EDGE_BUFFER);
 
+    const targetElementRightEdgeOffset =
+        window.innerWidth - targetElementDimension.left;
+
+    if (targetElementRightEdgeOffset < RIGHT_EDGE_BUFFER) {
```

Comment:
We are also not considering the available `window.scrollX` space that may be available like we do for stop edge.

Reply:
ok I will consider this as well. Will update code